SafeguardDevOpsService/Logic/SafeguardLogic.cs

```diff
@@ -116,6 +118,58 @@ private DevOpsException LogAndException(string msg, Exception ex = null)
                 AddDevOpsHeader(devOpsInstanceId, additionalHeaders), timeout);
         }
 
+        private byte[] IV =
+        {
+            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16
+        };
+        
+        private byte[] DeriveKeyFromPassPhrase(string passphrase)
+        {
+            const int iterations = 1000;
+            const int desiredKeyLength = 16; // 16 bytes equal 128 bits.
+            var emptySalt = Array.Empty<byte>();
+            var hashMethod = HashAlgorithmName.SHA384;
+            return Rfc2898DeriveBytes.Pbkdf2(Encoding.Unicode.GetBytes(passphrase),
+                emptySalt, iterations, hashMethod, desiredKeyLength);
+        }
+
+        private string Encrypt(string text, string passphrase)
+        {
+            // Don't do anything if the passphrase is null or empty
+            if (string.IsNullOrEmpty(passphrase))
+            {
+                return text;
+            }
+
+            var key = DeriveKeyFromPassPhrase(passphrase);
+            var iv = IV;
+
+            using var algorithm = Aes.Create();
+            var transform = algorithm.CreateEncryptor(key, iv);
+            var inputBuffer = Encoding.Unicode.GetBytes(text);
+            var outputBuffer = transform.TransformFinalBlock(inputBuffer, 0, inputBuffer.Length);
+            return Convert.ToBase64String(outputBuffer);
+        }
+
+        private string Decrypt(string text, string passphrase)
+        {
+            // Don't do anything if the passphrase is null or empty
+            if (string.IsNullOrEmpty(passphrase))
+            {
+                return text;
+            }
+
+            var key = DeriveKeyFromPassPhrase(passphrase);
+            var iv = IV;
+
+            using var algorithm = Aes.Create();
```

Comment:
Oh great, C# 8.0 now I have to re-learn everything!

Reply:
:)